osquery/tables/yara/yara.cpp

```diff
@@ -129,11 +147,20 @@ QueryData genYara(QueryContext& context) {
     for (const auto& group : groups) {
       if (rules.count(group) > 0) {
         doYARAScan(rules[group], path.c_str(), results, group, group);
+
+        // sleep between each file to help smooth out malloc spikes
+        std::this_thread::sleep_for(std::chrono::milliseconds(50));
       }
     }
   }
 
+#ifdef LINUX
+  if (osquery::FLAGS_yara_malloc_trim) {
+    malloc_trim(0);
```

Comment:
Should we instead run this after every query is executed?

Reply:
Not a bad idea.  To be conservative, I would propose we limit it to yara for users to get exposure to it, then expand it to the general schema at a future time.